modules/objdetect/include/opencv2/objdetect.hpp

```diff
@@ -305,6 +333,66 @@ class CV_EXPORTS_W CascadeClassifier
+    @param objects Vector of rectangles where each rectangle contains the detected object, the
+    rectangles may be partially outside the original image.
+    @param numDetections Vector of detection numbers for the corresponding objects. An object's number
+    of detections is the number of neighboring positively classified rectangles that were joined
+    together to form the object using non maxima suppression.
+    @param desiredScale Parameter specifying the scale on which the detection model needs to run.
+    @param minNeighbors Parameter specifying how many neighbors each candidate rectangle should have
+    to retain it.
+    @param flags Parameter with the same meaning for an old cascade as in the function
+    cvHaarDetectObjects. It is not used for a new cascade.
+    */
+    CV_WRAP_AS(detectSingleScale2) void detectSingleScale( InputArray image,
+                          CV_OUT std::vector<Rect>& objects,
+                          CV_OUT std::vector<int>& numDetections,
+                          Size desiredScale = Size(),
+                          int minNeighbors=3, int flags=0);
+    /** @brief Detects objects of a fixed size in the input image. The detected objects are returned as a list
+    of rectangles combined with their corresponding rejectlevel and levelweight.
+
+    @param image Matrix of the type CV_8U containing an image where objects are detected.
+    @param objects Vector of rectangles where each rectangle contains the detected object, the
+    rectangles may be partially outside the original image..
+    @param rejectLevels Parameter specifying the level at which the window was rejected. The last 4 stages are taken into account (hardcoded).
+    @param levelWeights Parameter specifying the final weight that lead to the decision at the rejected stage during the adaBoost process. It can be seen as a certainty of detection score. The higher the value, the more certain the detection is.
+    @param desiredScale Parameter specifying the scale on which the detection model needs to run.
+    @param minNeighbors Parameter specifying how many neighbors each candidate rectangle should have
+    to retain it.
+    @param flags Parameter with the same meaning for an old cascade as in the function
+    cvHaarDetectObjects. It is not used for a new cascade.
+    @param outputRejectLevels A boolean value, which needs to be set to true, in order to output the rejectlevels and levelweights.
+    */
+    CV_WRAP_AS(detectSingleScale3) void detectSingleScale( InputArray image,
+                          CV_OUT std::vector<Rect>& objects,
+                          CV_OUT std::vector<int>& rejectLevels,
+                          CV_OUT std::vector<double>& levelWeights,
+                          Size desiredScale = Size(),
+                          int minNeighbors = 3, int flags = 0,
+                          bool outputRejectLevels = false);
```

Comment:
Oh BTW, I completely understand why it is being used, so that the simply detectMultiScale can simply invoke the complexer one. But wouldn't it be more intuitive like it is now for the detectSingleScale? By default if you call the complex form, it is set to true. Other functions set it to false explicitly internally.

If I would change this for the detectMultiScale, would changing default parameter values break the ABI compatibility again? Or not? If not I am going to make the behaviour there also more straightforward and easier to understand.